subsys/net/ip/Kconfig

```diff
@@ -447,6 +447,37 @@ config NET_TCP_RECV_QUEUE_TIMEOUT
 	  SEQ 2. But if we receive SEQs 5,4,3,7 then the SEQ 7 is discarded
 	  because the list would not be sequential as number 6 is be missing.
 
+config NET_TCP_KEEPALIVE
+	bool "TCP Keep-Alives msg"
+	depends on NET_TCP
+	default y
```

Comment:
The RFC 1122 is very old one but it says that
```
4.2.3.6  TCP Keep-Alives

            Implementors MAY include "keep-alives" in their TCP
            implementations, although this practice is not universally
            accepted.  If keep-alives are included, the application MUST
            be able to turn them on or off for each TCP connection, and
            they MUST default to off.
```
so if we follow that the keepalive should be OFF by default. Just wondering should we have this off or on by default. Your use case probably is easier to configure if this is ON by default, or was there some other motivation for it?

Reply:
Setting NET_TCP_KEEPALIVE to ‘y’ just compiles the relevant code. To enable this function, you need to call the function setsockopt(sock, SOL_SOCKET, SO_KEEPALIVE, (void *)&keepalive, sizeof(keepalive)). So the default setting is ‘y’ to comply with the specification.